R/ml_evaluate.R

```diff
@@ -59,6 +74,30 @@ ml_evaluate.ml_generalized_linear_regression_model <- function(x, dataset) {
     new_ml_generalized_linear_regression_summary(fit_intercept = fit_intercept)
 }
 
+#' @rdname ml_evaluate
+#' @export
+ml_evaluate.ml_model_clustering <- function(x, dataset) {
+  if(inherits(x, "ml_model_lda")){
+    stop("`ml_evaluate()` is not supported for `ml_model_lda`.", call. = FALSE)
+  }
+
+  sc <- spark_connection(x$model)
+
+  if(spark_version(sc) >= "2.3.0"){
+    prediction <- x %>%
+      spark_jobj() %>%
+      invoke("transform", spark_dataframe(dataset))
+
+    silhouette <- sc %>%
+      invoke_new("org.apache.spark.ml.evaluation.ClusteringEvaluator") %>%
+      invoke("evaluate", prediction)
+
+    dplyr::tibble(Silhouette = silhouette)
+  } else{
+    warning("Silhouette is only available for spark 2.3.0 or greater.")
```

Comment:
```suggestion
    stop("Silhouette is only available for spark 2.3.0 or greater.")
```

Throw an error instead? To me it looks like `ml_evaluate` methods should either return a value or an error, so having a 3rd option of returning nothing is a bit unusual.

Reply:
I'd agree